images/win/scripts/Installers/Install-Ruby.ps1

```diff
@@ -0,0 +1,144 @@
+
+        }
+        return $versionLists
+    }
+    catch
+    {
+        Write-Host "Enable to send request to the '$uri'. Error: '$_'"
+        exit 1
+    }
+}
+
+# Most of this logic is from
+# https://github.com/MSP-Greg/actions-ruby/blob/master/lib/main.js
+function Install-Ruby
+{
+    param(
+        [String]$PackagePath,
+        [String]$Architecture = "x64"
+    )
+
+    # Expand archive with binaries
+    $packageName = [IO.Path]::GetFileNameWithoutExtension((Split-Path -Path $PackagePath -Leaf))
+    $tempFolder = Join-Path -Path $env:TEMP -ChildPath $packageName
+    Extract-7Zip -Path $PackagePath -DestinationPath $env:TEMP
+
+    # Get Ruby version from binaries
+    $rubyVersion = & "$tempFolder\bin\ruby.exe" -e "print RUBY_VERSION"
+
+    if ($rubyVersion)
+    {
+        Write-Host "Installing Ruby $rubyVersion"
+        $rubyToolcachePath = Join-Path -Path $env:AGENT_TOOLSDIRECTORY -ChildPath "Ruby"
+        $rubyVersionPath = Join-Path -Path $rubyToolcachePath -ChildPath $rubyVersion
+        $rubyArchPath = Join-Path -Path $rubyVersionPath -ChildPath $Architecture
+
+        if (-not (Test-Path $rubyToolcachePath))
+        {
+            Write-Host "Creating Ruby toolcache folder"
+            New-Item -ItemType Directory -Path $rubyToolcachePath | Out-Null
+        }
+
+        Write-Host "Creating Ruby '${rubyVersion}' folder in '${rubyVersionPath}'"
+        New-Item -ItemType Directory -Path $rubyVersionPath -Force | Out-Null
+
+        Write-Host "Moving Ruby '${rubyVersion}' files to '${rubyArchPath}'"
+        Move-Item -Path $tempFolder -Destination $rubyArchPath | Out-Null
+
+        Write-Host "Removing Ruby '${rubyVersion}' documentation '${rubyArchPath}\share\doc' folder"
+        Remove-Item -Path "${rubyArchPath}\share\doc" -Force -Recurse -ErrorAction Ignore
+
+        Write-Host "Installing Bundler"
+        cmd.exe /c "$rubyArchPath\bin\gem.cmd install bundler -v ""~> 2"" --no-document"
+
+        if ($LASTEXITCODE -ne 0)
+        {
+            Throw "Error happened during Ruby installation"
+            exit 1
+        }
+
+        Write-Host "Creating complete file"
+        New-Item -ItemType File -Path $rubyVersionPath -Name "$Architecture.complete" | Out-Null
+    }
+    else
+    {
+        Write-Host "Ruby application is not found. Failed to expand '$PackagePath' archive"
+        exit 1
+    }
+}
+
+function Set-DefaultRubyVersion
+{
+    param(
+        [Parameter(Mandatory=$true)]
+        [System.Version] $Version,
+        [System.String] $Arch = "x64"
+    )
+
+    $rubyPath = Join-Path $env:AGENT_TOOLSDIRECTORY "/Ruby/${Version}*/${Arch}/bin"
+    $rubyDir = (Get-Item -Path $rubyPath).FullName
+
+    Write-Host "Use Ruby ${Version} as a system Ruby"
+    Add-MachinePathItem -PathItem $rubyDir | Out-Null
+    $env:Path = Get-MachinePath
+
+    # Update ruby gem to latest version
+    Invoke-Expression "gem update --system --no-document"
```

Comment:
What's the motivation to update RubyGems? That might have various side effects, and make it harder for users to reproduce locally, as they typically have the version of RubyGems shipped with Ruby.

Also, `gem update --system` installs Bundler under `site_ruby`, which is rather confusing.

Reply:
Thank you. Removed.